README.md

```diff
@@ -28,25 +28,70 @@ public class ExampleIntegrationTest {
 ```
 
 #### @KnotxApplyConfiguration
-The annotation allows to specify one and more Knot.x configuration/s. It accepts a paths array and loads
-all configuration entries using [Vert.x Config](https://vertx.io/docs/vertx-config/java/) file stores. 
+
+The annotation allows to specify one and more Knot.x configuration(s) to load for given test.
+It accepts a paths array and loads all configuration entries using 
+[Vert.x Config](https://vertx.io/docs/vertx-config/java/) file stores (though through a custom 
+implementation which fixes a few things; see section [KnotxConcatConfigProcessor](#KnotxConcatConfigProcessor)). 
 It supports two configuration semantics:
 
 - JSON (files with `json` extension)
 - HOCON (files with `conf` extension)
 
-The order of the configuration files is important as it defines the overloading. 
-For conflicting key, configurations arriving last overloads the value provided by the previous 
-configuration files.
+The order of the configuration files is important, as it defines the overloading order 
+(base file first, overrides last).
+For conflicting (identical) keys in configuration, configuration entries arriving last 
+will overwrite the value provided by the previous configuration files.
 
 See [Vert.x Config overloading rules](https://vertx.io/docs/vertx-config/java/#_overloading_rules) 
 for more details.
 
+#### KnotxConcatConfigProcessor
+
+This implementation of Vert.x [ConfigProcessor](https://vertx.io/docs/vertx-config/java/#_extending_the_config_retriever)
+enables KnotxExtension to load Knot.x configuration files in a true hierarchical manner,
+including cross-file references to variables (only with HOCON files).
+ 
+From a user standpoint, HOCON cross-references should be available out-of-box in Vert.x's processing of configurations, since when you load configs
+programatically, you provide a source for given file and its format, and only JSON and HOCON are supported so it would be quite logical assumption.  
+This is not the case, however, and what Vert.x does internally is load all given files independently, evaluate them independently of each other,
```

Comment:
However, Vert.x loads all given files (stores), evaluates them independently and then stiches all files together to create a final result (JSON).

##### Cross-references example
We have two stores definitions:
```
"stores": [
      {
        "type": "file",
        "format": "hocon",
        "config": { "path": "config/application.conf" }
      },
      {
        "type": "json",
        "config": { "test.wiremock.mockService.port": 4321 }
      }
    ]
```
The `application.conf` file contains
```
config.somemodule.options.config {
  clientOptions {
    port = ${test.wiremock.mockService.port}
  }
}
test.wiremock {
  mockService {
    port = 1234
  }
}
```
The default Vert.x Config processing result:
```JSON
{
  "config": {
    "somemodule": {
      "options": {
        "config": {
          "clientOptions": {
            "port": 1234
          }
        }
      }
    }
  },
  "test": {
    "wiremock": {
      "mockService": {
        "port": 4321
      }
    }
  }
}
```
Expected behaviour:
```JSON
{
  "config": {
    "somemodule": {
      "options": {
        "config": {
          "clientOptions": {
            "port": 4321
          }
        }
      }
    }
  },
  "test": {
    "wiremock": {
      "mockService": {
        "port": 4321
      }
    }
  }
}
```